lib/live_view_native_swift_ui/types/color.ex

```diff
@@ -1,28 +1,64 @@
 defmodule LiveViewNativeSwiftUi.Types.Color do
+  @derive Jason.Encoder
   defstruct [
     :blue,
     :brightness,
-    :cg_color,
+    :create_with,
     :green,
     :hue,
-    :ns_color,
     :opacity,
     :red,
-    :rgb_color_space,
     :saturation,
     :string,
-    :type,
-    :ui_color,
     :white
   ]
 
+  @system_colors ~w(
+    black
+    blue
+    brown
+    clear
+    cyan
+    gray
+    green
+    indigo
+    mint
+    orange
+    pink
+    purple
+    red
+    teal
+    white
+    yellow
+    accent
+    primary
+    secondary
+  )a
+
   use LiveViewNativePlatform.Modifier.Type
   def type, do: :map
 
   def cast(value) when is_map(value), do: {:ok, struct(__MODULE__, value)}
-  def cast(value) when is_bitstring(value), do: {:ok, %__MODULE__{type: :string, string: value}}
-  def cast({:ui_color, ui_color}), do: {:ok, %__MODULE__{type: :ui_color, ui_color: ui_color}}
-  def cast({:ns_color, ns_color}), do: {:ok, %__MODULE__{type: :ns_color, ns_color: ns_color}}
-  def cast({:cg_color, cg_color}), do: {:ok, %__MODULE__{type: :cg_color, cg_color: cg_color}}
+
+  def cast(value) when is_bitstring(value),
+    do: {:ok, %__MODULE__{create_with: :string, string: value}}
+
+  def cast(value) when value in @system_colors,
+    do: {:ok, %__MODULE__{create_with: :string, string: "system-#{value}"}}
+
+  def cast({:rgb_color_space, %{white: white} = opts}),
```

Comment:
This naming doesn't feel right, since the color space is an option and this doesn't clearly show which color space it is. Could a color instead be represent with:

```ex
{:s_rgb, %{ red: 1, green: 0, blue: 0 }} # specify the sRGB space manually
{:display_p3, %{ red: 0, green: 1, blue: 1 }} # specify the displayP3 space manually
%{ red: 1, green: 0, blue: 0 } # defaults to be sRGB
```

Reply:
This is fixed. I went with the suggestion of passing the color space as the first element of the tuple, since the tuple form is only used in this case.